_posts/pl/2023-07-24-czy-wiesz-czym-jest-indexeddb.md

```diff
@@ -0,0 +1,210 @@
+---
+layout: post
+title: "IndexedDB - narzędzie do przechowywania danych w przeglądarce"
+date: 2023-07-24T11:00:00+01:00
+published: true
+didyouknow: false
+lang: pl
+author: wstolarski
+image: /assets/img/posts/2023-07-24-czy-wiesz-czym-jest-indexeddb/post_img.jpg
+description:
+tags:
+  - indexeddb
+  - webstorage
+  - database
+  - javascript
+---
+
+IndexedDB to wbudowana w przeglądarkę internetową baza danych typu NoSQL. Przechowuje ona dane lokalnie w przeglądarce,
+co pozwala na korzystanie z nich nawet wtedy, gdy urządzenie nie ma połączenia z internetem. Dzięki temu IndexedDB
+stanowi świetną opcję dla aplikacji internetowych i jest dobrą alternatywą dla Local Storage.
+
+W przeciwieństwie do Local Storage, dane w IndexedDB nie są automatycznie usuwane przez przeglądarkę. Zostają one
+przechowywane trwale do momentu, gdy użytkownik zdecyduje się usunąć je ręcznie lub gdy aplikacja, która korzysta z
+IndexedDB, wykona odpowiednie operacje usuwania.
+
+Dodatkowo IndexedDB pozwala na:
+
+- obsługę transakcji,
+- tworzenie zapytań typu range query,
+- obsługę indeksów,
+- przechowywanie znacznie większej ilości danych niż Local Storage i cookies.
+
+## Maksymalna ilość danych przechowywanych w IndexedDB
+Maksymalna ilość danych, którą można przechowywać w IndexedDB różni się w zależności od przeglądarki internetowej, a ograniczenia te mogą wynosić do kilku gigabajtów.
```

Comment:
Maksymalna ilość danych, którą można przechowywać w IndexedDB różni się w zależności od przeglądarki internetowej, ale zwykle jest to kilka gigabajtów.